client/src/SingleVideoCard.js

```diff
@@ -0,0 +1,71 @@
+import React, { useState } from "react";
+
+function SingleVideoCard({
+  videoId,
+  filterVideos,
+  setFilterVideos,
+  title,
+  url,
+}) {
+  const [count, setCount] = useState(0);
+
+  function minusCount() {
+    setCount((prevCount) => prevCount - 1);
+  }
+  function plusCount() {
+    setCount((prevCount) => prevCount + 1);
+  }
+
+  async function deleteVideo(event) {
+    try {
+      const response = await fetch(
+        `${process.env.REACT_APP_BACKEND_URL}/${videoId}`,
+        {
+          method: "DELETE",
+          headers: {
+            "Content-Type": "application/json",
+          },
+        }
+      );
+
+      if (response.ok) {
+        const updatedVideos = filterVideos.filter(
+          (video) => video.id !== videoId
+        );
+        setFilterVideos(updatedVideos);
+      } else {
+        console.error("Failed to delete video.");
+      }
+    } catch (error) {
+      console.error("Error deleting video:", error);
+    }
+  }
+
+  const urlId = url.split("v=")[1].substring(0, 11);
```

Comment:
Hi @christina-mifsud, 
Thanks for submitting your code - I can see you are trying to debug an error at the moment. Look closely at the data you're getting back from your API ( I can see this being logged in the `console` at the moment )
 
![image](https://github.com/CodeYourFuture/Full-Stack-Project-Assessment/assets/25401570/3c43e189-cac2-4b88-8a5a-41dfdb967f5e)

Reply:
If you look closely at this data does it make sense why you get a `TypeError` in the frontend when your application loads?

Reply:
Thank you for your comments Mitch!
I need to add validation to check if URL is a valid youtube URL or not?